app/gilded-rose.ts

```diff
@@ -1,69 +1,89 @@
+        if (backstagePasses.sellIn > 10) {
+            this.increaseQuality(backstagePasses);
+        } else if (backstagePasses.sellIn > 5 && backstagePasses.sellIn <= 10) {
+            this.increaseQuality(backstagePasses, 2);
+        } else if (backstagePasses.sellIn > 0 && backstagePasses.sellIn <= 5) {
+            this.increaseQuality(backstagePasses, 3);
+        } else {
+            backstagePasses.quality = 0;
         }
-      } else {
-        if (this.items[i].quality < 50) {
-          this.items[i].quality = this.items[i].quality + 1
-          if (this.items[i].name == 'Backstage passes to a TAFKAL80ETC concert') {
-            if (this.items[i].sellIn < 11) {
-              if (this.items[i].quality < 50) {
-                this.items[i].quality = this.items[i].quality + 1
-              }
+    }
+
+    dealWithConjuredItem(conjuredItem) {
+        if (conjuredItem.sellIn >= 0) {
+            this.decreaseQuality(conjuredItem, 2)
+        } else if (conjuredItem.sellIn < 0) {
+            this.decreaseQuality(conjuredItem, 4)
+        }
+    }
+
+    updateQuality() {
+        for (let i = 0; i < this.items.length; i++) {
+            if (this.items[i].name == 'Sulfuras, Hand of Ragnaros') {
+                continue;
             }
-            if (this.items[i].sellIn < 6) {
-              if (this.items[i].quality < 50) {
-                this.items[i].quality = this.items[i].quality + 1
-              }
+
+            this.items[i].sellIn -= 1;
+
+            if (this.items[i].name == 'Aged Brie') {
+                this.dealWithAgedBrie(this.items[i])
+                continue;
             }
-          }
-        }
-      }
-      if (this.items[i].name != 'Sulfuras, Hand of Ragnaros') {
-        this.items[i].sellIn = this.items[i].sellIn - 1;
-      }
-      if (this.items[i].sellIn < 0) {
-        if (this.items[i].name != 'Aged Brie') {
-          if (this.items[i].name != 'Backstage passes to a TAFKAL80ETC concert') {
-            if (this.items[i].quality > 0) {
-              if (this.items[i].name != 'Sulfuras, Hand of Ragnaros') {
-                this.items[i].quality = this.items[i].quality - 1
-              }
+
+            if (this.items[i].name == 'Backstage passes to a TAFKAL80ETC concert') {
+                this.dealWithBackstagePasses(this.items[i])
+                continue;
             }
-          } else {
-            this.items[i].quality = this.items[i].quality - this.items[i].quality
-          }
-        } else {
-          if (this.items[i].quality < 50) {
-            this.items[i].quality = this.items[i].quality + 1
-          }
+
+            if (this.items[i].name == 'Conjured Mana Cake') {
+                this.dealWithConjuredItem(this.items[i])
+                continue;
+            }
+
+            if (this.items[i].sellIn >= 0) {
+                this.decreaseQuality(this.items[i])
+            } else {
+                this.decreaseQuality(this.items[i], 2)
+            }
```

Comment:
If you're using `continue` in the previous ifs... You might as well use it here?